src/Validators/OptionalIfCheck.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Models;
+using form_builder.Models.Elements;
+
+namespace form_builder.Validators.IntegrityChecks.Form
+{
+    public class OptionalIfCheck : IFormSchemaIntegrityCheck
+    {
+        public IntegrityCheckResult Validate(FormSchema schema)
+        {
+            IntegrityCheckResult result = new();
+            List<string> questionIds = new();
+
+            List<IElement> elementsWithOptionalIfInJSON = schema.Pages
```

Comment:
Maybe go with the generic "elements"... the class itself should really say what's going on in here, or the Linq should be more readable and let other developers what's being returned. Don't think this variable needs to be that explicit.